lib/poller.js

```diff
@@ -216,9 +222,29 @@ var addMessage = module.exports.addMessage = function(message, callback) {
  *
  * @param  {Function}   callback    Standard callback function
  */
-var init = module.exports.init = function(callback) {
-  // Start polling the SQS queue
-  log.info('SQS polling initiated. Waiting for messages');
-  poll();
-  return callback();
-};
+ var init = module.exports.init = function(callback) {
+   // Start polling the SQS queue
+   var asyncTasks = [];
+   log.info('Asynchronous parallel SQS polling initiated. Waiting for messages');
+
+   // The max amount of messages that can be processed on a single receive is 10.
+   // Setting the concurrencyCount in the config helps to spin multiple pollers within the same deployment.
+   // Invoking more pollers helps in processing more messages. Also, provides the ability to throttle the lambda invokes.s
+   async.times(config.get('poller.concurrencyCount'), function(n, next) {
+     asyncTasks.push(function(callback) {
+       poll();
+     });
+     next();
+   }, function() {
+     async.parallel(asyncTasks, function(err) {
+       if (err) {
+         return callback(err);
+       }
+
+       log.info("Asynchronous parallel polling ended. Restarting poller");
+       init(function() { });
```

Comment:
@sandeepmjay FYI, I find `_.noop` a handy way to pass an empty function when needed
https://lodash.com/docs/4.17.4#noop